### PR TITLE
Pr/3078

### PR DESCRIPTION
**Pull request by jquense**

### CODE REVIEW

File: src/Modal.js [hunk 1]
```diff
@@ -149,147 +148,151 @@ class Modal extends React.Component {
...
+  handleDialogClick = e => {
+    if (e.target !== e.currentTarget) return;
 
     this.props.onHide();
-  }
+  };
+
+  handleEnter = (node, ...args) => {
+    if (node) {
+      node.style.display = 'block';
```
Comment (taion) [inline src/Modal.js]: so twiddling this is a new requirement with TWBS v4?

File: www/gatsby-node.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ exports.onCreateWebpackConfig = function onCreateWebpackConfig({
   getConfig
 }) {
   actions.setWebpackConfig({
+    devtool: 'cheap-inline-module-source-map',
```
Comment (taion) [inline www/gatsby-node.js]: remove?

File: src/ModalDialog.js [hunk 3]
```diff
@@ -2,61 +2,56 @@ import classNames from 'classnames';
...
+    dialogClassName: PropTypes.string,
+
+    /**
+     * Specifies a large or small modal.
+     *
+     * @type ('sm'|'lg')
+     */
+    size: PropTypes.string,
+
+    /**
```
Comment (taion) [inline src/ModalDialog.js]: sync up this comment with the one on `<Modal>`?

File: src/Modal.js (hunk 1 above)
Comment (jquense) [reply src/Modal.js]: This more correct for v3 as well but we don't allow modals to be in the DOM prior to showing so it's sort of irrelevant but I figured making it more correct can't hurt

File: src/ModalTitle.js [hunk 4]
```diff
@@ -1,31 +1,28 @@
...
+import { createBootstrapComponent } from './ThemeProvider';
 
-const propTypes = {
-  as: elementType
-};
-
-const defaultProps = {
-  as: 'h4'
-};
+const DivStyledAsH4 = divWithHeadingClass('h4');
```
Comment (taion) [inline src/ModalTitle.js]: so the idea is that this looks more or less per the examples, but doesn't insert a weird semantic header component? cool!

File: src/utils/divWithHeadingClass.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+import classNames from 'classnames';
+
+export default headingClass =>
```
Comment (taion) [inline src/utils/divWithHeadingClass.js]: arguably this should just be `divWithClassName` :p

File: src/Modal.js (hunk 1 above)
Comment (kt3k) [reply src/Modal.js]: @jquense @taion 
I think this change caused the issue https://github.com/react-bootstrap/react-bootstrap/issues/3399

`handleEnter` is called at onEnter of Transition object (in react-overlays/modal). When the user pass `animation={false}`, then the animation object is not created and therefore this method is not called.

**Pull request merged**